django/urls/converters.py

```diff
@@ -0,0 +1,91 @@
+import uuid
+
+from django.utils import lru_cache
+
+
+class BaseConverter(object):
+    pass
```

Comment:
What is the point in providing an empty base class? Is that in case we want to add some logic to all converters, both built-in and third-party, in the future?

Also you can drop `(object)` since the master branch no longer supports Python 3.

Reply:
Perhaps the base class could look like:

```python
class BaseConverter:

    @property
    def regex(self):
        raise NotImplementedError("BaseConverter subclasses must provide a regex")

    def to_python(self, value):
        return value

    def to_url(self, value):
        return value
```

to prevent duplicating these two methods in most subclasses.

Reply:
I'm a bit doubtful about providing default implementations of to_python/to_url: Yes, this implementation is used several times (so we could reduce the duplication). On the other hand: I'm not sure that's enough reason to just move the implementation to the base-class.

Also: I just noticed the name clashes with `django.utils.baseconv.BaseConverter`. Is that a problem? If so, maybe just remove the base-class?

Reply:
what about using BaseUrlConverter?

Reply:
Or should this be an abstract class with abstract methods?